### ISSUE TITLE
Datetime columns cause KeyError trying to access row 0

### ISSUE THREAD

**Issue by ilirmaci** [has2k1/plotnine#677]
Trying to cast strings to date on a view (or even a copy) of a `pandas.DataFrame` causes `KeyError: 0`. This error does not come up with the original column, but only after casting it to datetime.

Resetting the index on the subsetted data solves the problem.

```python
import numpy as np
import pandas as pd
from plotnine import ggplot, geom_point

large = pd.DataFrame({
    "date_str": 2*[f"2023-04-{x:02d}" for x in range(1, 31)],
    "category": 30*["a"] + 30*["b"],
    "value": np.random.normal(size=60)
})

small = large[large["category"] == "b"]    # view
small_copy = small.copy()                  # copy
small_new_index = small_copy.reset_index() # new index

# cast date string to a datetime object
small_copy["date_proper"] = pd.to_datetime(small_copy["date_str"])
small_new_index["date_proper"] = pd.to_datetime(small_new_index["date_str"])

# this works but the date is considered a category
ggplot(small_copy, aes(x='date_str', y='value')) + geom_point()

# this gives a KeyError, trying to access row 0
ggplot(small_copy, aes(x='date_proper', y='value')) + geom_point()

# this works because the index is reset
ggplot(small_new_index, aes(x='date_proper', y='value')) + geom_point()
```

**Comment by ilirmaci** [comment]
Duplicate of #642

**Comment by has2k1** [comment]
Run `pip install --upgrade mizani`.
